### PR TITLE
feat(post): creates the POST movies endpoint

### PR DESCRIPTION
**Pull request by forbaokhanh**
Creates the Movie database schema, model and controller as well as unit and integration tests.

### CODE REVIEW

File: lib/plugins/features/movies/controller.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const Movie = require('../../../models/movie');
+
+// defining code that is shared accross different files. Important to include so that the funciton or the variable can be used outside of this file
```
Comment (itsachen) [inline lib/plugins/features/movies/controller.js]: Nit: Remove note comments from the final PR. Generally we don't tend to comment too much of our production code unless it really needs clarification.

File: lib/plugins/features/movies/controller.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+exports.create = (payload) => {
+  // saves the apyload as the movie object in the database
+  return new Movie().save(payload).then( (movie) => {
```
Comment (itsachen) [inline lib/plugins/features/movies/controller.js]: Nit: Generally we define `then`s on newlines. So it would look something like this:
```js
return new Movie().save(payload)
.then((movie) => {
  // do something with movie
});
```

File: lib/plugins/features/movies/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const Controller = require('./controller');
+const MovieValidator = require('../../../validators/movie');
+
+exports.register = (server, options, next) => {
+
+	server.route([{
```
Comment (itsachen) [inline lib/plugins/features/movies/index.js]: Nit: Indenting looks off here

File: lib/plugins/features/movies/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      }
+    }
+  }]);
+
+  next();
+};
+
+exports.register.attributes = {
+  name: 'movies'
+};
```
Comment (itsachen) [inline lib/plugins/features/movies/index.js]: Nit: Newline at the end of the file

File: lib/server.js [hunk 5]
```diff
@@ -13,7 +13,12 @@ const server = new Hapi.Server({
 server.connection({ port: Config.PORT });
 
 server.register([
-  require('hapi-bookshelf-serializer')
+  require('hapi-bookshelf-serializer'),
+  {
+    register: require('hapi-format-error'),
+      options: { joiStatusCode: 422 }
```
Comment (itsachen) [inline lib/server.js]: Nit: Indents look off here

File: test/plugins/features/movies/controller.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+        return new Movie({ id: movie.id}).fetch();
+      })
+      .then((movie) => {
+        expect(movie.get('title')).to.eql(payload.title);
+      });
+    });
+
+  });
+});
```
Comment (itsachen) [inline test/plugins/features/movies/controller.test.js]: Nit: Generally we leave a newline after the first line and before the last line of a `describe` block, similar to how you arranged the `create` block

File: test/plugins/features/movies/controller.test.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+const Controller = require('../../../../lib/plugins/features/movies/controller');
+const Movie = require('../../../../lib/models/movie');
+
+describe('movie controller', () => {
+
+  describe('create', () => {
+
+    it ('creates a movie', () => {
+      const payload = { title: 'WALL-E'};
```
Comment (itsachen) [inline test/plugins/features/movies/controller.test.js]: Nit: One-line object literals should be padded by spaces, so something like this:
```js
{ title: 'WALL-E' }
```

File: test/plugins/features/movies/controller.test.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+      return Controller.create(payload).then((movie) => {
```
Comment (itsachen) [inline test/plugins/features/movies/controller.test.js]: Nit: We generally put `.then` on a new line

File: test/plugins/features/movies/controller.test.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        expect(movie.get('title')).to.eql(payload.title);
+
+        return new Movie({ id: movie.id}).fetch();
```
Comment (itsachen) [inline test/plugins/features/movies/controller.test.js]: Nit: Add a space to even out padding in object literal

File: test/plugins/features/movies/index.test.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+      })
+      .then((response) => {
+        expect(response.statusCode).to.eql(200);
+        expect(response.result.object).to.eql('movie');
+      });
+    });
+
+  });
+
+});
```
Comment (itsachen) [inline test/plugins/features/movies/index.test.js]: Nit: Add newline here

File: test/validators/movie.test.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+const Joi = require('joi');
+const MovieValidator = require('../../lib/validators/movie');
+
+describe('movie validator', () => {
+
+  describe('title', () => {
+
+    it('is required', () => {
+      // create a test payload
```
Comment (itsachen) [inline test/validators/movie.test.js]: Nit: Remove note comments (for this file)

File: test/validators/movie.test.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+      };
+      const result = Joi.validate(payload, MovieValidator);
+
+      expect(result.error.details[0].path).to.eql('release_year');
+      expect(result.error.details[0].type).to.eql('number.max');
+    });
+
+  });
+
+});
```
Comment (itsachen) [inline test/validators/movie.test.js:54]: Nit: Add a newline

**Review by itsachen — APPROVED** [review]
LGTM! Just some syntax/style nits.

File: lib/plugins/features/movies/controller.js (hunk 2 above)
Comment (sjl2) [reply lib/plugins/features/movies/controller.js]: Also in this case, you can use the arrow notation that is just 
```
() => <something to return>
```
More specifically,
```
.then((movie) => new Movie({ id: movie.id }).fetch())
```

File: lib/validators/movie.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+const Joi = require('joi');
+
+module.exports = Joi.object().keys({
+  title: Joi.string().min(1).max(255).required(),
```
Comment (sjl2) [inline lib/validators/movie.js]: nit: I don't think `.min(1)` is necessary here since `Joi.string()` doesn't allow the empty string. Or that may be the case on API.

**Review by sjl2 — APPROVED** [review]
LGTM with a couple nits and Anthony's suggestions. You could also add a test to the title validator tests that checks that it doesn't allow empty string if there was concern about leaving out the `.min(1)`.
